sound/soc/soc-pcm.c

```diff
@@ -1415,37 +1415,42 @@ int dpcm_path_get(struct snd_soc_pcm_runtime *fe,
 	return paths;
 }
 
+static bool dpcm_be_is_active(struct snd_soc_dpcm *dpcm, int stream,
+			      struct snd_soc_dapm_widget *widget,
+			      struct snd_soc_dapm_widget_list *list)
+{
+	struct snd_soc_dai *dai;
+	unsigned int i;
+
+	/* prune the BE if it's no longer in our active list */
+	if (widget && widget_in_list(list, widget))
+		return true;
```

Comment:
@lyakh I think we can split this patch into smaller bits to make review easier like this part where you move dpcm_be_is_active() out of dpcm_prune_paths().

But, is returning true correct here? If you return true here, then "if (!force && !dpcm_be_is_active(dpcm, stream, widget, *list_))" on line 1453 will be false and you will not continue, right?

Reply:
@ranj063 splitting - can do. Returning true - that's a rebasing bug, thanks for spotting!